src/Factory/ConfigInjector.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-tooling for the canonical source repository
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-tooling/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Zend\Expressive\Tooling\Factory;
+
+/**
+ * Inject factory configuration in an autoloadable location.
+ *
+ * This class will re-generate the file denoted by its CONFIG_FILE constant.
+ * It first pulls in the data in that file, if the file exists, and then adds
+ * an entry for the given class, pointing it to the given factory, rewriting
+ * the configuration file on completion.
+ */
+class ConfigInjector
+{
+    public const CONFIG_FILE = 'config/autoload/zend-expressive-tooling-factories.global.php';
```

Comment:
Is there a reason why a new file is used and not `config/autoload/dependencies.global.php`?

Reply:
Yes: to prevent issues when re-generating the file:

- serialization of closures
- resolution of constants
- expressions
- usage of import statements

It's far more complex to write to a file _meant_ to be managed manually than to use a file that we can mark as _under control of the tooling_.

Reply:
Makes sense. Thanx for the explanation.